### PR TITLE
lib/{storage,flagutil} - Add option for snapshot autoremoval

### PR DESCRIPTION
**Pull request by tenmozes**
- add prometheus-like duration as command flag
- add option to delete stale snapshots
- update duration.go flag to re-use own code

### CODE REVIEW

File: app/vmstorage/main.go [hunk 1]
```diff
@@ -374,6 +376,22 @@ func RequestHandler(w http.ResponseWriter, r *http.Request) bool {
 	}
 }
 
+func deleteStaleSnapshots() {
```
Comment (f41gh7) [inline app/vmstorage/main.go:379]: I think, that this function must be added to the storage and included into storage watchers, like  freeDiskSpaceWatcher for example:
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/lib/storage/storage.go#L756

File: lib/flagutil/duration.go [hunk 2]
```diff
@@ -12,14 +12,14 @@ import (
 // NewDuration returns new `duration` flag with the given name, defaultValue and description.
 //
 // DefaultValue is in months.
-func NewDuration(name string, defaultValue float64, description string) *Duration {
+func NewDuration(name string, defaultValue string, description string) *Duration {
```
Comment (f41gh7) [inline lib/flagutil/duration.go:15]: I don't think, that we should change API for exist flag or it should be done via separate PR.

File: app/vmstorage/main.go [hunk 3]
```diff
@@ -22,10 +22,11 @@ import (
 )
 
 var (
-	retentionPeriod   = flagutil.NewDuration("retentionPeriod", 1, "Data with timestamps outside the retentionPeriod is automatically deleted")
+	retentionPeriod   = flagutil.NewDuration("retentionPeriod", "1", "Data with timestamps outside the retentionPeriod is automatically deleted")
 	snapshotAuthKey   = flag.String("snapshotAuthKey", "", "authKey, which must be passed in query string to /snapshot* pages")
 	forceMergeAuthKey = flag.String("forceMergeAuthKey", "", "authKey, which must be passed in query string to /internal/force_merge pages")
 	forceFlushAuthKey = flag.String("forceFlushAuthKey", "", "authKey, which must be passed in query string to /internal/force_flush pages")
+	snapshotsMaxAge   = flagutil.NewPromDuration("snapshotsMaxAge", "1w", "Snapshots outside the snapshotsMaxAge are automatically deleted. Check happens every hour.")
```
Comment (f41gh7) [inline app/vmstorage/main.go:29]: It's better to disable it by default, we shouldn't change exist behavior. It may confuse users.

 WDYT about moving `snapshot autoremoval` feature into `vmbackupmanager`? 

 It can retrieve list of snapshots by API call and remove stalesnapshots by configured  maxAge.

**Review by valyala — APPROVED** [review]

**Comment by valyala** [comment]
Thanks!
